LiskHQ/lisk
# Improve logging of WebSocket messages and events - Closes #1762

### What was the problem?

The current node doesn't have sufficient low-level WebSocket logging to aid in QA debugging.

### How did I fix it?

Added logging (debug/trace) for all major WebSocket events and all WebSocket messages for both inbound and outbound sockets.

### How to test it?

- In `config.json` set the `consoleLogLevel` to `trace`.
- Run the node.

There should now be additional `[trc]` and `[dbg]` log messages which look like this:
- `[Inbound socket] WebSocket handshake from socket MqHahyi-hAfHYXU3AAAD failed with code 4101: Expected nonce to be not equal to: FvPC4PCawdOUm31G`
- `[Outbound socket] Peer message received: {"event":"...","data":...}`

`[Inbound socket]` means that the connection which triggered the issue was created/initiated by a peer. `[Outbound socket]` means that the connection which triggered the issue was created/initiated by the current node.

Also, you can run the test:
`npm run mocha test/functional/ws/transport/client.js`

### Review checklist

* The PR solves #1762
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated
